### PR TITLE
bugfix: LT-2278 - Advice entities

### PR DESCRIPTION
**Pull request by andytaylorkainos**

### CODE REVIEW

File: cases/helpers.py [hunk 1]
```diff
@@ -132,13 +132,48 @@ def create_grouped_advice(case, request, advice, level):
...
+
+    final_advice = FinalAdvice.objects.distinct(*FinalAdvice.ENTITIES).filter(case=case)
+
+    if advice_type:
+        final_advice = final_advice.filter(type=advice_type)
+
+    final_advice_entities = defaultdict(list)
+
+    for advice in final_advice:
+        for entity_name in FinalAdvice.ENTITIES:
```
Comment (a-bek) [inline cases/helpers.py:169]: Why not use the entity property on the Advice model we created below?

Comment (andytaylorkainos) [reply cases/helpers.py:169]: Because then i lose the ability to map the entity to different serializers :/

Comment (a-bek) [reply cases/helpers.py:169]: Could map it by object type returned by entity?

File: cases/helpers.py [hunk 2]
```diff
@@ -132,13 +132,45 @@ def create_grouped_advice(case, request, advice, level):
...
+        final_advice = final_advice.filter(type=advice_type)
+
+    final_advice_entities = defaultdict(list)
+
+    for advice in final_advice:
+        serializer = entity_field_serializer_map[advice.entity_field]
+        data = serializer(advice.entity).data
+        final_advice_entities[advice.entity_field].append(data)
+
+    return final_advice_entities
```
Comment (BenAndrew-Kainos) [inline cases/helpers.py:173]: Does this function return items in lists only for objects which there may be multiple (i.e. goods)

File: cases/managers.py [hunk 3]
```diff
@@ -245,3 +246,21 @@ def create(self):
...
         case_reference_code.save()
         return case_reference_code
+
+
+class AdviceManager(models.Manager):
+    def get_queryset(self):
+        return super().get_queryset().prefetch_related(*self.model.ENTITY_FIELDS)
+
+    def get(self, *args, **kwargs):
+        entity_id = kwargs.pop("entity_id", None)
```
Comment (BenAndrew-Kainos) [inline cases/managers.py:256]: Could you add a docstring for this function & its usage

File: cases/helpers.py (hunk 2 above)
Comment (andytaylorkainos) [reply cases/helpers.py:135]: returns a dictionary containing lists as the values

File: cases/managers.py (hunk 3 above)
Comment (andytaylorkainos) [reply cases/managers.py:260]: done!

**Pull request merged**